x/compute/internal/keeper/keeper.go

```diff
@@ -124,37 +124,56 @@ func GetSignerSignature(signer exported.Account, tx auth.StdTx) (authtypes.StdSi
 	return authtypes.StdSignature{}, fmt.Errorf("could not find signer signature")
 }
 
+func (k Keeper) GetSignerInfo(ctx sdk.Context, signer sdk.AccAddress) (authtypes.StdSignature, []byte, error) {
+	// Warning: This API may be deprecated:
+	// https://github.com/cosmos/cosmos-sdk/commit/c13809062ab16bf193ad3919c77ec03c79b76cc8#diff-a64b9f4b7565560002e3ac4a5eac008bR148
+	tx := authtypes.StdTx{}
+	txBytes := ctx.TxBytes()
+	err := k.cdc.UnmarshalBinaryLengthPrefixed(txBytes, &tx)
+	if err != nil {
+		return authtypes.StdSignature{
+			PubKey:    secp256k1.PubKeySecp256k1{},
+			Signature: []byte{},
+		}, nil, sdkerrors.Wrap(types.ErrInstantiateFailed, fmt.Sprintf("Unable to decode transaction from bytes: %s", err.Error()))
+	}
```

Comment:
To make these slightly nicer, you can assign the empty `authtypes.StdSignature` to a local in the beginning of the function, and just refer to it in the returns.